### PR TITLE
[risk=low][no ticket] Refactor inst admin edit tests, add a few, and fix a minor bug

### PR DESCRIPTION
**Pull request by jmthibault79**
Following up on a comment in #5639 as well as some general cleanup to support other imminent changes to inst admin edit

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/admin/admin-institution-edit.spec.tsx [hunk 1]
```diff
@@ -13,6 +13,30 @@ import {simulateComponentChange, waitOneTickAndUpdate} from 'testing/react-test-
 import {InstitutionApiStub, VERILY, VERILY_WITHOUT_CT} from 'testing/stubs/institution-api-stub';
 import {AdminInstitutionEdit} from './admin-institution-edit';
 
+const findRTDetails = (wrapper) => wrapper.find('[data-test-id="registered-card-details"]');
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-institution-edit.spec.tsx:16]: most changes (but not all) in this file relate to these wrapper.find() helpers

File: ui/src/app/pages/admin/admin-institution-edit.spec.tsx [hunk 2]
```diff
@@ -514,64 +467,151 @@ describe('AdminInstitutionEditSpec - add mode', () => {
     expect(wrapper).toBeTruthy();
   });
 
+  it('should throw error for a new Institution if the display name is more than 80 characters', async() => {
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-institution-edit.spec.tsx:470]: this new test failed (why I didn't include it in the other PR)

but the fix was easy: see below.

File: ui/src/app/pages/admin/admin-institution-edit.spec.tsx [hunk 3]
```diff
@@ -514,64 +467,151 @@ describe('AdminInstitutionEditSpec - add mode', () => {
...
+    findRTAddressInput(wrapper).first().simulate('change', {target: {value: 'user@domain.com'}});
+    findRTAddressInput(wrapper).first().simulate('blur');
 
     // RT no change
-    expect(wrapper.find('[data-test-id="registered-email-address-input"]').first().prop('value'))
-        .toBe('user@domain.com');
+    expect(findRTAddressInput(wrapper).first().prop('value')).toBe('user@domain.com');
+  });
+
+  it('should not change eRA Required and tier enabled switches when changing tier requirement', async() => {
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-institution-edit.spec.tsx:539]: these new Add Mode tests were copied from Edit Mode and lightly edited, passing without app-code changes.

File: ui/src/app/pages/admin/admin-institution-edit.tsx [hunk 4]
```diff
@@ -672,7 +672,7 @@ export const AdminInstitutionEdit = fp.flow(withNavigation, withRouter)(class ex
                 onBlur={v => this.setState(fp.set(['institution', 'displayName'], v.trim()))}
             />
             <div style={{color: colors.danger}} data-test-id='displayNameError'>
-              {!this.isAddInstitutionMode && errors && errors.displayName}
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-institution-edit.tsx:675]: this was the minor bug: did not complain when the displayName was too long in Add Mode.

Comment (jmthibault79) [reply ui/src/app/pages/admin/admin-institution-edit.tsx:675]: The Save button was correctly disabled.

File: ui/src/app/pages/admin/admin-institution-edit.spec.tsx [hunk 5]
```diff
@@ -514,64 +467,151 @@ describe('AdminInstitutionEditSpec - add mode', () => {
...
+    const wrapper = component();
+    await waitOneTickAndUpdate(wrapper);
+    expect(wrapper).toBeTruthy();
+
+    await simulateComponentChange(wrapper, findCTEnabled(wrapper), true);
+    await waitOneTickAndUpdate(wrapper);
+
+    await simulateComponentChange(wrapper, findCTDropdown(wrapper), InstitutionMembershipRequirement.DOMAINS);
+
+    // Single Entry with incorrect Email Domain format
```
Comment (calbach) [inline ui/src/app/pages/admin/admin-institution-edit.spec.tsx:609]: nit: is this comment correct? There's just an empty line, right?

**Review by calbach — APPROVED** [review]

File: ui/src/app/pages/admin/admin-institution-edit.spec.tsx (hunk 5 above)
Comment (jmthibault79) [reply ui/src/app/pages/admin/admin-institution-edit.spec.tsx]: right - about half of these "incorrect format" are extraneous empty line

Comment (jmthibault79) [reply ui/src/app/pages/admin/admin-institution-edit.spec.tsx]: updated some comments

**Review by calbach — APPROVED** [review]

**Pull request merged**